### PR TITLE
Topic/v1 notes

### PR DESCRIPTION
**Pull request by amosr**

### CODE REVIEW

**Comment by tranma** [comment]
can haz merge this https://github.com/ambiata/icicle/pull/194 into this branch?

**Comment by amosr** [comment]
please do :+1:

**Comment by amosr** [comment]
can I merge this?

**Comment by tranma** [comment]
:spaghetti:

File: src/Icicle/Sea/FromAvalanche/Program.hs [hunk 1]
```diff
@@ -245,8 +245,39 @@ seaOfXValue v
...
+     VArray vs
+      | ArrayT t' <- t
+      -> let len = length vs
+             writes arr (v',i)
+              = prim (PrimUpdate $ PrimUpdateArrayPut t')
+                     [arr, int i, seaOfXValue v' t']
+             init
+              = prim (PrimUnsafe $ PrimUnsafeArrayCreate t')
+                     [int len]
+        in  foldl writes init (vs `List.zip` [0..])
```
Comment (amosr) [inline src/Icicle/Sea/FromAvalanche/Program.hs]: have a look at this if you don't mind. it should end up as something like
```
 [ "thing" ]
 iarray_put( iarray_create(s->mempool, 1), 0, "thing" )
```

Comment (tranma) [inline src/Icicle/Sea/FromAvalanche/Program.hs]: I haven't seen this in action in the REPL yet, what's an example?

Comment (amosr) [inline src/Icicle/Sea/FromAvalanche/Program.hs]: actually I think it might only be required for maps, after we split maps into two arrays

Comment (tranma) [inline src/Icicle/Sea/FromAvalanche/Program.hs]: also when you read a buffer using `buf_read` (I'll do buffers and we'll see)

Comment (amosr) [inline src/Icicle/Sea/FromAvalanche/Program.hs]: I don't think it will be required for that actually - this is only for compiletime constants

**Pull request merged**